Add pipeline build-govsvc-docker

philandstuff opened the pull request:
The intention for this pipeline is to build former `gdsre` docker hub
organisation images on concourse rather than ad hoc on people's
laptops.

Currently it pushes to ECR rather than docker hub as a proof of
concept, but if people think it's a sensible thing to do, we can a)
change it to push to docker hub (involves adding docker hub creds to
the pipeline), b) add all the other `gdsre` dockerfiles too.

We're moving to the `govsvc` docker hub org because we already have a bot user set up to push there.

philandstuff committed:
Add pipeline build-gdsre-docker

The intention for this pipeline is to build `gdsre` docker hub
organisation images on concourse rather than ad hoc on people's
laptops.

Currently it pushes to ECR rather than docker hub as a proof of
concept, but if people think it's a sensible thing to do, we can a)
change it to push to docker hub (involves adding docker hub creds to
the pipeline), b) add all the other `gdsre` dockerfiles too.

chrisfarms commented on reliability-engineering/pipelines/build-gdsre-docker.yml:
```diff
@@ -0,0 +1,37 @@
+- name: gdsre-aws-ruby-git
+  type: git
+  source:
+    uri: https://github.com/alphagov/verify-infrastructure.git
```
should we move the Dockerfile into this repo? ... seems weird being over there given it's name and generic use

philandstuff replied:
yes we should, but one step at a time

philandstuff replied:
ie not in this PR

philandstuff replied:
actually I've convinced myself otherwise, it'll be easier to move now than when the pipeline is active

philandstuff committed:
Move gdsre/aws-ruby dockerfile to this repo

philandstuff committed:
Make pipeline self-updating

philandstuff committed:
Push to docker hub

This also changes to the `govsvc` docker hub org, because we already
have creds set up for pushing to that org, while we don't for gdsre.
Probably better to consolidate on fewer docker hub orgs anyway.

adityapahuja commented on reliability-engineering/pipelines/build-govsvc-docker.yml:
```diff
@@ -0,0 +1,52 @@
+  source:
+    username: ((dockerhub-username))
+    password: ((dockerhub-password))
+    repository: govsvc/aws-ruby
```
What will happen to https://hub.docker.com/r/gdsre/aws-ruby?

philandstuff replied:
we'll migrate away from it, change usages to this one instead
